### PR TITLE
Fix out-of-memory condition in conf

### PR DESCRIPTION
**Pull request by mattcaswell**
conf has the ability to expand variables in config files. Repeatedly doing
this can lead to an exponential increase in the amount of memory required.
This places a limit on the length of a value that can result from an
expansion.

Credit to OSS-Fuzz for finding this problem.

### CODE REVIEW

**Comment by levitte** [comment]
Dunno, really. I don't like putting this arbitrary limit on the value length. 

It's interesting, though, that it's possible to have an unnamed key...

**Comment by mattcaswell** [comment]
> I don't like putting this arbitrary limit on the value length

Yeah, I know. I tried to set it large enough that it really shouldn't make any practical difference though.

**Comment by richsalz** [comment]
I think 8K is more than enough.  But the limit should be documented.

**Comment by richsalz** [comment]
And "unnamed key" sounds like a bug IMO

**Comment by mattcaswell** [comment]
Updated with some documentation. @levitte - are you objecting to this, or can I take an approval from @richsalz?

**Comment by mattcaswell** [comment]
Also - which branches do you think this should go to? The issue exists in all branches, but not sure it is worth backporting it.

**Comment by mattcaswell** [comment]
I upped the limit to 64k at the suggestion of @vdukhovni to account for "...any plausible jumbo
hex-encoded DER blobs for inclusion in certificates". @richsalz still ok?

**Comment by richsalz** [comment]
sure

**Comment by levitte** [comment]
Thank you, the higher limit made this change much more acceptable to me.

**Comment by levitte** [comment]
And oh, if it needs to be said: all branches

**Comment by mattcaswell** [comment]
Pushed. Thanks.

File: fuzz/corpora/conf/0d7ad6e04c0235cdc590756ceec867a05cff5823 [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+=$$$$$$$
+=$$$
+=$󠁝$$
+=$$$
+=$$$
+=$$$
+=$$$
+=$$$
+=$$$
+=$$$$$
```
Comment (vdukhovni) [inline fuzz/corpora/conf/0d7ad6e04c0235cdc590756ceec867a05cff5823]: FWIW, there appears to be a missing newline here.  Is that intentional?

Comment (mattcaswell) [inline fuzz/corpora/conf/0d7ad6e04c0235cdc590756ceec867a05cff5823]: There are a few strange quirks in this file (there are some invisible non-printed characters in there as well). Its just the auto-generated fuzz file created by oss-fuzz. I just inserted it as-is without any amendments.

**Pull request closed without merging**